atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -250,8 +303,21 @@ def resolve_signing_intent(self):
         self.compose_config.set_signing_intent(signing_intent['name'])
 
     def forward_composes(self):
-        yum_repourls = [compose_info['result_repofile'] for compose_info in self.composes_info]
-        override_build_kwarg(self.workflow, 'yum_repourls', yum_repourls)
+        repos_by_arch = {}
+        # set overrides by arch if arches are available
+        for compose_info in self.composes_info:
+            for arch in compose_info.get('arches', []):
+                repos_by_arch.setdefault(arch, [])
+                repos_by_arch[arch].append(compose_info['result_repofile'])
+
+        if repos_by_arch:
```

Comment:
This is not necessary, you can just jump into the for-loop, it'l be a noop if dict is empty.

Reply:
for else: conditionals look weird to me.

Reply:
Fixed.